src/curl.rs

```diff
@@ -17,8 +17,18 @@ where
         .arg(&url)
         .output()
         .map(move |result| match result {
-            Ok(output) => f(&output.stdout),
-            Err(err) => Err(Error::from(err)),
+            Ok(output) => {
+                if output.status.success() {
```

Comment:
Can you use more idiomatic rust case if expression here?

also not quite a fan of the custom error. But it does give more context than before